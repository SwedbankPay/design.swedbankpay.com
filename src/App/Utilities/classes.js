export const spacingClasses = {
	margins: [
		{
			name: "m-0",
			css: "0",
			px: " ",
		},
		{
			name: "m-1",
			css: "0.25rem",
			px: "4px",
		},
		{
			name: "m-2",
			css: "0.5rem",
			px: "8px",
		},
		{
			name: "m-3",
			css: "1rem",
			px: "16px",
		},
		{
			name: "m-4",
			css: "1.5rem",
			px: "24px",
		},
		{
			name: "m-5",
			css: "3rem",
			px: "48px",
		},
	],

	paddings: [
		{
			name: "p-0",
			css: "0",
			px: " ",
		},
		{
			name: "p-1",
			css: "0.25rem",
			px: "4px",
		},
		{
			name: "p-2",
			css: "0.5rem",
			px: "8px",
		},
		{
			name: "p-3",
			css: "1rem",
			px: "16px",
		},
		{
			name: "p-4",
			css: "1.5rem",
			px: "24px",
		},
		{
			name: "p-5",
			css: "3rem",
			px: "48px",
		},
	],
};

export const borderClasses = [
	{
		name: "border",
		css: "border: 1px solid @light-brown",
	},
	{
		name: "border-top",
		css: "border-top: 1px solid @light-brown",
	},
	{
		name: "border-bottom",
		css: "border-bottom: 1px solid @light-brown",
	},
	{
		name: "border-left",
		css: "border-left: 1px solid @light-brown",
	},
	{
		name: "border-right",
		css: "border-right: 1px solid @light-brown",
	},
];

export const breakpointClasses = [
	{
		name: "xs",
		css: "0px",
		px: ".col-8 .d-block",
	},
	{
		name: "sm",
		css: "576px",
		px: ".col-sm-8 .d-sm-block",
	},
	{
		name: "md",
		css: "768px",
		px: ".col-md-8 .d-md-block",
	},
	{
		name: "lg",
		css: "992px",
		px: ".col-lg-8 .d-lg-block",
	},
	{
		name: "xl",
		css: "1200px",
		px: ".col-xl-8 .d-xl-block",
	},
	{
		name: "xxl",
		css: "1600px",
		px: ".col-xxl-8 .d-xxl-block",
	},
];

export const hidingElementClasses = {
	hidingElements: [
		{
			screenSize: "Hidden on all",
			dClass: "d-none",
		},
		{
			screenSize: "Hidden only on xs",
			dClass: "d-none d-sm-block",
		},
		{
			screenSize: "Hidden only on md",
			dClass: "d-md-none d-lg-block",
		},
		{
			screenSize: "Hidden only on xl",
			dClass: "d-xl-none d-xxl-block",
		},
		{
			screenSize: "Visible on all",
			dClass: "d-block",
		},
		{
			screenSize: "Visible only on sm",
			dClass: "d-sm-block d-md-none",
		},
		{
			screenSize: "Visible only on lg",
			dClass: "d-lg-block d-xl-none",
		},
		{
			screenSize: "Visble only on xxl",
			dClass: "d-xxl-block",
		},
	],

	notation: [
		{
			value: "d-none",
			css: "display: none",
		},
		{
			value: "d-inline",
			css: "display: inline",
		},
		{
			value: "d-inline-block",
			css: "display: inline-block",
		},
		{
			value: "d-block",
			css: "display: block",
		},
		{
			value: "d-table",
			css: "display: table",
		},
		{
			value: "d-table-cell",
			css: "display: table-cell",
		},
		{
			value: "d-table-row",
			css: "display: table-row",
		},
		{
			value: "d-flex",
			css: "display: flex",
		},
		{
			value: "d-inline-flex",
			css: "display: inline-flex",
		},
		{
			value: "d-grid",
			css: "display: grid",
		},
		{
			value: "d-inline-grid",
			css: "display: inline-grid",
		},
	],
};

export const flexClasses = {
	flexBehaviour: [
		{
			code: "d-flex",
		},
		{
			code: "d-inline-flex",
		},
		{
			code: "d-{breakpoint}-flex",
		},
		{
			code: "d-{breakpoint}-inline-flex",
		},
	],

	direction: [
		{
			code: "flex-row",
		},
		{
			code: "flex-row-reverse",
		},
		{
			code: "flex-column",
		},
		{
			code: "flex-column-reverse",
		},
		{
			code: "flex-{breakpoint}-row",
		},
		{
			code: "flex-{breakpoint}-row-reverse",
		},
		{
			code: "flex-{breakpoint}-column",
		},
		{
			code: "flex-{breakpoint}-column-reverse",
		},
	],

	justifyContent: [
		{
			code: "justify-content-start",
		},
		{
			code: "justify-content-end",
		},
		{
			code: "justify-content-center",
		},
		{
			code: "justify-content-between",
		},
		{
			code: "justify-content-around",
		},
		{
			code: "justify-content-evenly",
		},
		{
			code: "justify-content-{breakpoint}-start",
		},
		{
			code: "justify-content-{breakpoint}-end",
		},
		{
			code: "justify-content-{breakpoint}-center",
		},
		{
			code: "justify-content-{breakpoint}-between",
		},
		{
			code: "justify-content-{breakpoint}-around",
		},
		{
			code: "justify-content-{breakpoint}-evenly",
		},
	],

	alignItems: [
		{
			code: "align-items-start",
		},
		{
			code: "align-items-end",
		},
		{
			code: "align-items-baseline",
		},
		{
			code: "align-items-stretch",
		},
		{
			code: "align-items-{breakpoint}-start",
		},
		{
			code: "align-items-{breakpoint}-end",
		},
		{
			code: "align-items-{breakpoint}-center",
		},
		{
			code: "align-items-{breakpoint}-baseline",
		},
		{
			code: "align-items-{breakpoint}-stretch",
		},
	],

	alignSelf: [
		{
			code: "align-self-start",
		},
		{
			code: "align-self-end",
		},
		{
			code: "align-self-center",
		},
		{
			code: "align-self-baseline",
		},
		{
			code: "align-self-stretch",
		},
		{
			code: "align-self-{breakpoint}-start",
		},
		{
			code: "align-self-{breakpoint}-end",
		},
		{
			code: "align-self-{breakpoint}-center",
		},
		{
			code: "align-self-{breakpoint}-baseline",
		},
		{
			code: "align-self-{breakpoint}-stretch",
		},
	],

	fill: [
		{
			code: "flex-fill",
		},
		{
			code: "flex-{breakpoint}-fill",
		},
	],

	growAndShrink: [
		{
			code: "flex-{grow|shrink}-0",
		},
		{
			code: "flex-{grow|shrink}-1",
		},
		{
			code: "flex-{breakpoint}-{grow|shrink}-0",
		},
		{
			code: "flex-{breakpoint}-{grow|shrink}-1",
		},
	],

	wrap: [
		{
			code: "flex-wrap",
		},
		{
			code: "flex-nowrap",
		},
		{
			code: "flex-wrap-reverse",
		},
	],

	order: [
		{
			code: "order-{0-12}",
		},
		{
			code: "order-{breakpoint}-{0-12}",
		},
	],

	alignContent: [
		{
			code: "align-content-start",
		},
		{
			code: "align-content-end",
		},
		{
			code: "align-content-center",
		},
		{
			code: "align-content-around",
		},
		{
			code: "align-content-stretch",
		},
		{
			code: "align-content-{breakpoint}-start",
		},
		{
			code: "align-content-{breakpoint}-end",
		},
		{
			code: "align-content-{breakpoint}-center",
		},
		{
			code: "align-content-{breakpoint}-around",
		},
		{
			code: "align-content-{breakpoint}-stretch",
		},
	],
};

export const imageClasses = [
	{
		code: "img-fluid",
		explaination: "Scales to parent",
	},
	{
		code: "img-rounded",
		explaination: "Rounded corners",
	},
	{
		code: "img-circle",
		explaination: "Perfect circle",
	},
	{
		code: "img-thumbnail",
		explaination: "Adds border and padding",
	},
];

export const sizingClasses = [
	{
		code: "{w|h}-25",
	},
	{
		code: "{w|h}-50",
	},
	{
		code: "{w|h}-75",
	},
	{
		code: "{w|h}-100",
	},
];

export const textClasses = [
	{
		code: "text-justify",
	},
	{
		code: "text-left",
	},
	{
		code: "text-center",
	},
	{
		code: "text-right",
	},
	{
		code: "text-{breakpoint}-left",
	},
	{
		code: "text-nowrap",
	},
	{
		code: "text-truncate",
	},
	{
		code: "text-lowercase",
	},
	{
		code: "text-uppercase",
	},
	{
		code: "text-capitalize",
	},
	{
		code: "font-weight-bold",
	},
	{
		code: "font-weight-normal",
	},
	{
		code: "font-weight-light",
	},
	{
		code: "font-italic",
	},
];

export const visibilityClasses = [
	{
		code: "visible",
	},
	{
		code: "invisible",
	},
];
