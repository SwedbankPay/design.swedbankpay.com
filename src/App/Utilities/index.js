import React from "react";

import { RenderPage } from "../utils";

const Core = () => (
    <RenderPage path="/utilities" initPath="/utilities/utilities" />
);

export default Core;
