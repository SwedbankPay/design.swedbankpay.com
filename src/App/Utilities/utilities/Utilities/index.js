import React from "react";
import { Link } from "react-router-dom";
import overviewList from "@src/App/routes/utilities";

import { DocContainer } from "@docutils";

const MenuOverview = () => (
    <section className="component-overview hide-arrow-icon">
        {overviewList[0].routes.slice(1).map(card => (
            <Link key={card.title} to={card.path} className="cards cards-primary cards-wide">
                <div className="cards-content m-0">
                    <span className="h3">{card.title}</span>
                    <span>{card.text}</span>
                </div>
                <i className="material-icons">arrow_forward</i>
            </Link>
        ))}
    </section>
);

const Utilities = () => (
    <DocContainer>
        <p className="lead">
        Utilities are simple CSS classes that can be used to streamline the process of applying general styling and to override a style already defined in the component CSS.
        </p>
        <MenuOverview />
    </DocContainer>
);

export default Utilities;

export { MenuOverview };
