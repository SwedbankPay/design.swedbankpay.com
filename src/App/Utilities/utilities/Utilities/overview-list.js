const overviewList = [
    {
        path: "/utilities/borders",
        title: "Borders",
        text: "Set borders that suit your needs"
    },
    {
        path: "/utilities/breakpoints",
        title: "Breakpoints",
        text: "See our display breakpoints"
    },
    {
        path: "/utilities/colors",
        title: "Colors",
        text: "See our color utility classes"
    },
    {
        path: "/utilities/display",
        title: "Display",
        text: "See our display properties"
    },
    {
        path: "/utilities/flex",
        title: "Flex",
        text: "Layout, alignment and grid"
    },
    {
        path: "/utilities/images",
        title: "Images",
        text: "Modify how images are displayed"
    },
    {
        path: "/utilities/sizing",
        title: "Sizing",
        text: "Size according to your needs"
    },
    {
        path: "/utilities/spacing",
        title: "Spacing",
        text: "Responsive margin and padding"
    },
    {
        path: "/utilities/text",
        title: "Text",
        text: "Control wrapping, weight and more"
    },
    {
        path: "/utilities/visibility",
        title: "Visibility",
        text: "Control visibilities of display"
    }
];

export default overviewList;
