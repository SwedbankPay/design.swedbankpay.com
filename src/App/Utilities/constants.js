import React from "react";
import { spacingClasses, borderClasses, breakpointClasses, hidingElementClasses, flexClasses, imageClasses, sizingClasses, textClasses, visibilityClasses } from "./classes";
import CodeTagComponent from "@components/CodeTags";

const UtilityPresenter = ({ util, css, example, breakpointVersion }) => (
    <div className="d-flex justify-content-between mb-2">
        <CodeTagComponent code={util} type="primary"/>
        <CodeTagComponent code={css} type="secondary"/>
        {breakpointVersion ? <CodeTagComponent type="tertiary" code={example}/> : <>{example && <span>{example}</span>}</>}
    </div>
);

const DisplayPresenter = ({ util, css, textVersion }) => (
    <div className="d-flex justify-content-between mb-2">
        {textVersion ? <span>{util}</span> : <CodeTagComponent code={util} type="primary"/>}
        <CodeTagComponent code={css} type="secondary"/>
    </div>
);

const FlexPresenter = ({ code }) => (
    <div className="d-flex  mb-2">
        <CodeTagComponent code={code} type="primary"/>
    </div>
);

export const borders = [
    {
        id: "borders",
        title: "Border",
        subtitle: "Gives borders to an element.",
        content: <>
            {borderClasses.map((rule, i) => (
                <UtilityPresenter key={rule + i} util={rule.name} css={rule.css}/>
            ))}
        </>
    }];

export const breakpoints = [
    {
        id: "breakpoints",
        title: "Breakpoints",
        subtitle: "Break it up",
        content: <>
            <div className="d-flex justify-content-between">
                <p className="font-weight-bold">Infix</p>
                <p className="font-weight-bold">Min-width</p>
                <p className="font-weight-bold">Example use</p>
            </div>
            {breakpointClasses.map((rule, i) => (
                <UtilityPresenter breakpointVersion key={rule + i} util={rule.name} css={rule.css} example={rule.px}/>
            ))}
        </>
    }
];

export const spacings = [
    {
        id: "margins",
        title: "Margin",
        subtitle: "For seperating elements.",
        content: <>
            {spacingClasses.margins.map((rule, i) => (
                <UtilityPresenter key={rule + i} util={rule.name} css={rule.css} example={rule.px}/>
            ))}
        </>
    },
    {
        id: "paddings",
        title: "Padding",
        subtitle: "Give the element some space.",
        content: <>
            {spacingClasses.paddings.map((rule, i) => (
                <UtilityPresenter key={rule + i} util={rule.name} css={rule.css} example={rule.px}/>
            ))}
        </>
    }
];

export const colors = [
    {
        id: "colorText",
        title: "Text",
        subtitle: "To alter text colors",
        content: <>
            <p className="text-default ">.text-default</p>
            <p className="text-muted ">.text-muted</p>
            <p className="text-white text-dark-background ">.text-white</p>
            <p className="text-danger ">.text-danger</p>
            <p className="text-warning ">.text-warning</p>
            <p className="text-neutral ">.text-neutral</p>
            <p className="text-success ">.text-success</p>
        </>
    },
    {
        id: "colorBackground",
        title: "Background colors",
        subtitle: "To alter background colors of an element",
        content: <>
            <div className="text-default bg-brand p-1">.bg-brand</div>
            <div className="col-12 text-default bg-primary p-1">.bg-primary</div>
            <div className="col-12 text-default bg-secondary p-1">.bg-secondary</div>
            <div className="col-12 text-default bg-tertiary p-1">.bg-tertiary</div>
            <div className="col-12 text-default bg-quaternary p-1">.bg-quaternary</div>
            <div className="col-12 text-default bg-danger p-1">.bg-danger</div>
            <div className="col-12 text-default bg-warning p-1">.bg-warning</div>
            <div className="col-12 text-default bg-success p-1">.bg-success</div>
            <div className="col-12 text-default bg-neutral p-1">.bg-neutral</div>
        </>
    }
];

export const displayHidingElements = [
    {
        id: "displaynotation",
        title: "Notation",
        subtitle: "Quick display classes. d(isplay)-<value>",
        content: <>
            {hidingElementClasses.notation.map((rule, i) => (
                <DisplayPresenter key={rule + i} util={rule.value} css={rule.css}/>
            ))}
        </>
    },
    {
        id: "hidingElmenents",
        title: "Hiding elements",
        subtitle: "For faster mobile-friendly development",
        content: <>
            {hidingElementClasses.hidingElements.map((rule, i) => (
                <DisplayPresenter key={rule + i} util={rule.screenSize} css={rule.dClass} textVersion/>
            ))}
        </>
    }
];

export const flex = [
    {
        id: "enableflexbahaviour",
        title: "Enable flex bahaviour",
        subtitle: "For creating a flexbox",
        content: <>
            {flexClasses.flexBehaviour.map((rule, i) => (
                <FlexPresenter key={rule + i} code={rule.code} explaination={rule.explaination}/>
            ))}
        </>
    },
    {
        id: "flexDirection",
        title: "Direction",
        subtitle: "Direct the flex",
        content: <>
            {flexClasses.direction.map((rule, i) => (
                <FlexPresenter key={rule + i} code={rule.code} explaination={rule.explaination}/>
            ))}
        </>
    },
    {
        id: "justifyContent",
        title: "Justify content",
        subtitle: "Justify your flex contents",
        content: <>
            {flexClasses.justifyContent.map((rule, i) => (
                <FlexPresenter key={rule + i} code={rule.code} explaination={rule.explaination}/>
            ))}
        </>
    },
    {
        id: "alignItems",
        title: "Align items",
        subtitle: "Align set of elements",
        content: <>
            {flexClasses.alignItems.map((rule, i) => (
                <FlexPresenter key={rule + i} code={rule.code} explaination={rule.explaination}/>
            ))}
        </>
    },
    {
        id: "alignSelf",
        title: "Align self",
        subtitle: "Align element",
        content: <>
            {flexClasses.alignSelf.map((rule, i) => (
                <FlexPresenter key={rule + i} code={rule.code} explaination={rule.explaination}/>
            ))}
        </>
    },
    {
        id: "fill",
        title: "Fill",
        subtitle: "Flex fill",
        content: <>
            {flexClasses.fill.map((rule, i) => (
                <FlexPresenter key={rule + i} code={rule.code} explaination={rule.explaination}/>
            ))}
        </>
    },
    {
        id: "growAndShrink",
        title: "Grow and shrink",
        subtitle: "Effective use of space",
        content: <>
            {flexClasses.growAndShrink.map((rule, i) => (
                <FlexPresenter key={rule + i} code={rule.code} explaination={rule.explaination}/>
            ))}
        </>
    },
    {
        id: "wrap",
        title: "Wrap",
        subtitle: "Nowrap is a browsers defualt",
        content: <>
            {flexClasses.wrap.map((rule, i) => (
                <FlexPresenter key={rule + i} code={rule.code} explaination={rule.explaination}/>
            ))}
        </>
    },
    {
        id: "order",
        title: "Order",
        subtitle: "Modifies the visual order",
        content: <>
            {flexClasses.order.map((rule, i) => (
                <FlexPresenter key={rule + i} code={rule.code} explaination={rule.explaination}/>
            ))}
        </>
    },
    {
        id: "alignContent",
        title: "Align content",
        subtitle: "No effect on single rows of flex items",
        content: <>
            {flexClasses.alignContent.map((rule, i) => (
                <FlexPresenter key={rule + i} code={rule.code} explaination={rule.explaination}/>
            ))}
        </>
    }
];

export const images = [
    {
        id: "images",
        title: "Images",
        subtitle: "Utility classes to your images",
        content: <>
            {imageClasses.map((rule, i) => (
                <DisplayPresenter key={rule + i} util={rule.explaination} css={rule.code} textVersion/>
            ))}
        </>
    }
];

export const sizings = [
    {
        id: "sizing",
        title: "Sizing",
        subtitle: "Resize your elements",
        content: <>
            {sizingClasses.map((rule, i) => (
                <FlexPresenter key={rule + i} code={rule.code}/>
            ))}
        </>
    }
];

export const text = [
    {
        id: "text",
        title: "Text",
        subtitle: "Modify your text elements",
        content: <>
            {textClasses.map((rule, i) => (
                <FlexPresenter key={rule + i} code={rule.code}/>
            ))}
        </>
    }
];

export const visibility = [
    {
        id: "visibility",
        title: "Visibility",
        subtitle: "Modify the visual representation of your elements",
        content: <>
            {visibilityClasses.map((rule, i) => (
                <FlexPresenter key={rule + i} code={rule.code}/>
            ))}
        </>
    }
];
