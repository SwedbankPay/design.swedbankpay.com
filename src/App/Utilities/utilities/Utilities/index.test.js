import React from "react";
import { shallow } from "enzyme";

import Utilities, { MenuOverview } from "./index";

describe("Utilities: Utilities", () => {
    it("is defined", () => {
        expect(Utilities).toBeDefined();
    });

    it("renders", () => {
        const wrapper = shallow(<Utilities />);

        expect(wrapper).toMatchSnapshot();
    });

    describe("MenuOverview", () => {
        it("is defined", () => {
            expect(MenuOverview).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<MenuOverview />);

            expect(wrapper).toMatchSnapshot();
        });
    });
});
