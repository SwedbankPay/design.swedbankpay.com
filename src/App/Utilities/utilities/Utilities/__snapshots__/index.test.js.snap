// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Utilities: Utilities MenuOverview renders 1`] = `
<section
  className="component-overview hide-arrow-icon"
>
  <Link
    className="cards cards-primary cards-wide"
    key="Borders"
    to="/utilities/borders"
  >
    <div
      className="cards-content m-0"
    >
      <span
        className="h3"
      >
        Borders
      </span>
      <span>
        Set borders that suit your needs
      </span>
    </div>
    <i
      className="material-icons"
    >
      arrow_forward
    </i>
  </Link>
  <Link
    className="cards cards-primary cards-wide"
    key="Breakpoints"
    to="/utilities/breakpoints"
  >
    <div
      className="cards-content m-0"
    >
      <span
        className="h3"
      >
        Breakpoints
      </span>
      <span>
        See our display breakpoints
      </span>
    </div>
    <i
      className="material-icons"
    >
      arrow_forward
    </i>
  </Link>
  <Link
    className="cards cards-primary cards-wide"
    key="Colors"
    to="/utilities/colors"
  >
    <div
      className="cards-content m-0"
    >
      <span
        className="h3"
      >
        Colors
      </span>
      <span>
        See our color utility classes
      </span>
    </div>
    <i
      className="material-icons"
    >
      arrow_forward
    </i>
  </Link>
  <Link
    className="cards cards-primary cards-wide"
    key="Display"
    to="/utilities/display"
  >
    <div
      className="cards-content m-0"
    >
      <span
        className="h3"
      >
        Display
      </span>
      <span>
        See our display properties
      </span>
    </div>
    <i
      className="material-icons"
    >
      arrow_forward
    </i>
  </Link>
  <Link
    className="cards cards-primary cards-wide"
    key="Flex"
    to="/utilities/flex"
  >
    <div
      className="cards-content m-0"
    >
      <span
        className="h3"
      >
        Flex
      </span>
      <span>
        Layout, alignment and grid
      </span>
    </div>
    <i
      className="material-icons"
    >
      arrow_forward
    </i>
  </Link>
  <Link
    className="cards cards-primary cards-wide"
    key="Images"
    to="/utilities/images"
  >
    <div
      className="cards-content m-0"
    >
      <span
        className="h3"
      >
        Images
      </span>
      <span>
        Modify how images are displayed
      </span>
    </div>
    <i
      className="material-icons"
    >
      arrow_forward
    </i>
  </Link>
  <Link
    className="cards cards-primary cards-wide"
    key="Sizing"
    to="/utilities/sizing"
  >
    <div
      className="cards-content m-0"
    >
      <span
        className="h3"
      >
        Sizing
      </span>
      <span>
        Size according to your needs
      </span>
    </div>
    <i
      className="material-icons"
    >
      arrow_forward
    </i>
  </Link>
  <Link
    className="cards cards-primary cards-wide"
    key="Text"
    to="/utilities/text"
  >
    <div
      className="cards-content m-0"
    >
      <span
        className="h3"
      >
        Text
      </span>
      <span>
        Control wrapping, weight and more
      </span>
    </div>
    <i
      className="material-icons"
    >
      arrow_forward
    </i>
  </Link>
  <Link
    className="cards cards-primary cards-wide"
    key="Visibility"
    to="/utilities/visibility"
  >
    <div
      className="cards-content m-0"
    >
      <span
        className="h3"
      >
        Visibility
      </span>
      <span>
        Control visibilities of display
      </span>
    </div>
    <i
      className="material-icons"
    >
      arrow_forward
    </i>
  </Link>
</section>
`;

exports[`Utilities: Utilities renders 1`] = `
<DocContainer>
  <p
    className="lead"
  >
    Utilities are simple CSS classes that can be used to streamline the process of applying general styling and to override a style already defined in the component CSS.
  </p>
  <MenuOverview />
</DocContainer>
`;
